.github/get-modified-packages.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * Given a list of all packages, find the package that have been modified.
+ */
+
+if (3 > $_SERVER['argc']) {
+    echo "Usage: app-packages modified-files\n";
+    exit(1);
+}
+
+$allPackages = json_decode($_SERVER['argv'][1], true, 512, \JSON_THROW_ON_ERROR);
+$modifiedFiles = json_decode($_SERVER['argv'][2], true, 512, \JSON_THROW_ON_ERROR);
+
+function isComponentBridge(string $packageDir): bool
+{
+    return 0 < preg_match('@Symfony/Component/.*/Bridge/@', $packageDir);
+}
+
+$newPackage = [];
+$modifiedPackages = [];
+foreach ($modifiedFiles as $file) {
+    foreach ($allPackages as $package) {
+        if (0 === strpos($file, $package)) {
+            $modifiedPackages[$package] = true;
+            if ('LICENSE' === substr($file, -7)) {
+                /*
+                 * There is never a reason to modify the LICENSE file, this diff
```

Comment:
not even the yearly bump of the year in the license file ?

Reply:
The yearly bump is wrong. =)
If we insist of doing it we are okey with this job failing once per year. 

-----------

The Copyright year should be the year the work is first created. There is no need to update it every year nor to have a `from-to` pattern. Sure, I'm no layer and I don't find the legal references I once read so it is no real point of having a discussion about it now. 

But let's think about it. If you were required to update the year on your copyright claim, that would allow a bad person just "steal" the work no matter of copyright the first of January (just before the "yearly bump"). And when you sue them for using copyrighted work the judge will just say: "The work was stolen the first of January and you didn't update the copyright string until the 3rd of January." and then close the case...

=)

Just because half the internet is doing something, it doesn't mean that way is correct.

Reply:
I'm not a lawyer and I did not spend time on it, but my understanding was that we need to add the current year whenever we modify the file. So, using 2000-2020 is not good if there were no changes in 2010 (as you mention, code written before 2010 is still protected anyway), we should use 2000-2009,2011-2020. Anyway, I'm not sure we should care that much here.